테스트 코드 수정완료

pangpangE123 opened the pull request:
통합테스트 이후 단위테스트들 수정된 코드에 맞게 다시 수정했습니다 
검토부탁드립니다!

pangpangE123 committed:
commonResponse 적용,
adaptor에서 로그인 요청 변경
login -> /auth/api/login

pangpangE123 committed:
테스트 코드 모두 정상작동하도록 수정

ckddms6530 commented:
빌드가 되면 리뷰 하도록 하겠습니다!

pangpangE123 committed:
미등록된 commonResponse 등록

AoiTuNa commented:
이거 커밋 master로 되어 있어요 develop로 바꿔야 할거 같은데여!

pangpangE123 commented:
> 이거 커밋 master로 되어 있어요 develop로 바꿔야 할거 같은데여!

앗 죄송합니다 변경했어요!

AoiTuNa commented on src/main/java/com/nhnacademy/front/server/service/AuthService.java:
```diff
@@ -18,7 +18,7 @@ public interface AuthService {
      * @param password 유저가 입력한 password 값 입니다!
      * @return 인증된 유저의 accessToken을 JwtToken으로 래핑해서 리턴합니다. 만약 없다면 null을 반환합니다!
```
p4:이거 설명 수정좀 해주세요! JwtToken ->String

AoiTuNa replied:
수정했습니다

AoiTuNa commented on src/main/java/com/nhnacademy/front/server/adapter/impl/AuthAdapterImpl.java:
```diff
@@ -34,13 +36,14 @@ public JwtToken userLogin(String id, String password) {
+        }
     );
-    return exchange.getBody();
+    return exchange.getBody().dataOrElseThrow(() -> new RuntimeException("로그인 실패"));
```
p4: 이거 실패 하게 두면 바로 예외 처리로 빠져버리나요? 아니면 login화면으로 redirect 시키는 건가요?
auth Service 부분에 RuntimeException을 catch하나요?

AoiTuNa replied:
해당 예외를 catch 하여 null을 반환하고 controller 단에서 null에 대한 처리를 추가해서 조치하였습니다

AoiTuNa committed:
수정사항 수정
커스텀 exception LoginFailedException 추가
던진 예외를 잡으면 null을 반환해 controller 단에서 null처리 추가
javadoc 설정 수정

AoiTuNa reviewed (commented):
수정 완료

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/adapter/impl/AuthAdapterImpl.java:
```diff
@@ -34,13 +38,14 @@ public JwtToken userLogin(String id, String password) {
-    ResponseEntity<JwtToken> exchange = restTemplate.exchange(
-        "http://192.168.0.27:8080/login",
+    ResponseEntity<CommonResponse<LoginResponseDto>> exchange = restTemplate.exchange(
+        "http://192.168.71.99:8080/api/auth/login",
```
[P3] : restTemplate 에서 openFeign 처럼 유레카에서 다른 서비스의 Ip와 port를 서비스 이름으로 가져올 수 있습니다 방법 찾아보시고 적용해보세요!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/domain/LoginResponseDto.java:
```diff
@@ -4,16 +4,17 @@
 /**
- * 유저의 accessToken의 정보가 들어있는 domain Class 입니다!<p></p>
- * JsonProperty : "access_token" 입니다!
+ * 유저의 accessToken의 정보가 들어있는 domain Class 입니다!<p></p> JsonProperty : "access_token" 입니다!
```
[P3] : 이 부분은 이제 지워도 될 것 같습니다~

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/service/impl/AuthServiceImpl.java:
```diff
@@ -18,10 +19,15 @@ public class AuthServiceImpl implements AuthService {
+        LoginResponseDto loginResponseDto;
+        try{
+            loginResponseDto = authAdapter.userLogin(id,password);
+        }catch (LoginFailedException e){
```
[P4] : 이 부분에서 잡지않고 어드바이스로 넘겨서 공통적으로 처리해도 괜찮지 않을까요?

ckddms6530 commented on src/test/java/com/nhnacademy/front/server/adapter/impl/AuthAdapterImplTest.java:
```diff
@@ -34,13 +36,18 @@ public void setUp() {
   @Test
   void userLogin() {
-    mockServer.expect(MockRestRequestMatchers.requestTo("http://192.168.0.27:8080/login"))
+    mockServer.expect(MockRestRequestMatchers.requestTo("http://192.168.71.99:8080/api/auth/login"))
```
[P3] : 테스트 코드에 @DisplayName 어노테이션으로 무슨 테스트인지 작성해주시면 다른 분들도 보기 편할 것 같습니다!
